docs/proposals/choosing-scheduler.md

```diff
@@ -0,0 +1,161 @@
+	// The rule defined by Policy is applied if the pod satisfies all of the criteria in PolicyPredicate
+	// (currently this just means satisfying the PodSelector in PolicyPredicate).
+	// The reason we use PolicyPredicate instead of just putting *PodSelector here is to allow
+	// us to cleanly add additional predicates in the future; they would be added to
+	// PolicyPredicate, allowing PodPolicyRule to stay simple.
+	PolicyPredicate *PolicyPredicate `json:"policyPredicate,omitempty"`
+	Policy string `json:"policy"`
+}
+
+// Note that Namespace is not part of PolicyPredicate because we assume pod policies are per-Namespace.
+type PolicyPredicate struct {
+	PodSelector *PodSelector `json:"podSelector,omitempty"`
+}
+```
+
+### PodSchedulerPolicy admission controller
+
+```go
+// PodSchedulerPolicySpec defines the policy configuration for the PodSchedulerPolicy admission controller.
+type PodSchedulerPolicySpec struct {
+	Policy PodPolicy `json:"policy,omitempty"`
+}
+
+// PodSchedulerPolicyStatus just reflects back the policy set in the spec.
+type PodSchedulerPolicyStatus struct {
+	Policy PodPolicy `json:"policy,omitempty"`
+}
+
+// PodSchedulerPolicy describes the PodSchedulerPolicy admission controller,
+// which maps pods to the scheduler responsible for scheduling them.
+type PodSchedulerPolicy struct {
+	unversioned.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the pod scheduler policy implemented by this admission controller.
+	// http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status
+	Spec PodSchedulerPolicySpec `json:"spec,omitempty"`
+
+	// Status reflects back the spec.
+	// http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status
+	Status PodSchedulerPolicyStatus `json:"status,omitempty"`
+}
+
+// PodSchedulerPolicyList is a list of PodSchedulerPolicy items.
+type PodSchedulerPolicyList struct {
+	unversioned.TypeMeta `json:",inline"`
+	// Standard list metadata.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#types-kinds
+	unversioned.ListMeta `json:"metadata,omitempty"`
+
+	// Items is a list of PodSchedulerPolicy objects.
+	// More info: http://releases.k8s.io/HEAD/docs/design/admission_control_resource_quota.md#admissioncontrol-plugin-resourcequota
+	Items []PodSchedulerPolicy `json:"items"`
+}
+```
+
+## Implementation plan
+
+1. Create the `PodPolicy` API resource
+1. Create the PodSchedulerPolicy admission controller
+
+## Future work
+
+In the future we may add more pod properties to `PolicyPredicate`. We considered allowing
+`PolicyPredicate` to look at pod resource requirements and limits, so it could direct pods to
+schedulers based on their QoS class (best-effort, burstable, guaranteed). We might do this in
+the future, but for now we will have the admission controller set a label
```

Comment:
Admission controllers can update pods, for example limitranger can update resource requirements. So I think it should be possible for an admission controller to set label.